tensorflow/python/ops/nn_ops.py

```diff
@@ -2202,9 +2202,9 @@ def conv1d(value, filters, stride, padding,
       the filter is moved right at each step.
     padding: 'SAME' or 'VALID'
     use_cudnn_on_gpu: An optional `bool`.  Defaults to `True`.
-    data_format: An optional `string` from `"NHWC", "NCHW"`.  Defaults
-      to `"NHWC"`, the data is stored in the order of
-      [batch, in_width, in_channels].  The `"NCHW"` format stores
+    data_format: An optional `string` from `"NWC", "NCW"`.  Defaults
```

Comment:
We cannot change the signature this way, as @jhseu said, this is a backwards incompatible change. Instead, you could accept both NHWC and NWC. You can even only document NWC to push users away from NHWC. Optionally, you can use @deprecated_arg_values to mark NWHC explicitly deprecated (will trigger a warning if used).

Reply:
Thanks @martinwicke @jhseu for the review. The current PR actually accept both `NWC` and `NHWC`, (see Ln 2219):
```
    if data_format is None or data_format == "NHWC" or data_format == "NWC":
      data_format = "NHWC"
      spatial_start_dim = 1
      strides = [1, 1, stride, 1]
    elif data_format == "NCHW" or data_format == "NCW":
      data_format = "NCHW"
      spatial_start_dim = 2
      strides = [1, 1, 1, stride]
    else:
```

I think `@deprecated_arg_values` is good. Will update the PR to add it.